catalog/core/catalog-core-solrcommands/src/main/java/org/codice/ddf/commands/solr/RestoreCommand.java

```diff
@@ -0,0 +1,319 @@
+      uriBuilder.addParameter("command", "restore");
+
+      if (StringUtils.isNotBlank(backupLocation)) {
+        uriBuilder.addParameter("location", backupLocation);
+      }
+
+      URI restoreUri = uriBuilder.build();
+      LOGGER.debug("Sending request to {}", restoreUri);
+
+      printResponse(sendGetRequest(restoreUri));
+    } catch (IOException | SolrServerException e) {
+      LOGGER.info("Unable to perform single node Solr restore, core: {}", coreName, e);
+    }
+  }
+
+  private void performSolrCloudRestore() throws IOException {
+    SolrClient client = null;
+    try {
+      client = getCloudSolrClient();
+      if (status) {
+        verifyStatusInput();
+        printStatus(client, requestId);
+      } else {
+        verifyRestoreInput();
+        performSolrCloudRestore(client);
+      }
+    } finally {
+      shutdown(client);
+    }
+  }
+
+  private void printResponse(@Nullable HttpResponse response) {
+    if (response != null) {
+      StatusLine statusLine = response.getStatusLine();
+      if (statusLine.getStatusCode() == HttpStatus.SC_OK) {
+        printSuccessMessage(String.format("%nRestore of [%s] complete.%n", coreName));
+      } else {
+        printErrorMessage(String.format("Error restoring Solr core: [%s]", coreName));
+        printErrorMessage(
+            String.format(
+                "Restore command failed due to: %d - %s",
+                statusLine.getStatusCode(), statusLine.getReasonPhrase()));
+        printErrorMessage(String.format("Request: %s", getReplicationUrl(coreName)));
+      }
+    }
+  }
+
+  private boolean restore(
+      SolrClient client, String collection, String backupLocation, String backupName)
+      throws IOException, SolrServerException {
+    if (canRestore(client, collection)) {
+      CollectionAdminRequest.Restore restore =
+          CollectionAdminRequest.AsyncCollectionAdminRequest.restoreCollection(
+                  collection, backupName)
+              .setLocation(backupLocation);
+
+      String syncReqId = restore.processAsync(client);
+
+      while (true) {
+        CollectionAdminRequest.RequestStatusResponse requestStatusResponse =
+            CollectionAdminRequest.requestStatus(syncReqId).process(client);
+        RequestStatusState requestStatus = requestStatusResponse.getRequestStatus();
+        if (requestStatus == RequestStatusState.COMPLETED) {
+          LOGGER.debug("Restore status: {}", requestStatus);
+          return true;
+        } else if (requestStatus == RequestStatusState.FAILED
+            || requestStatus == RequestStatusState.NOT_FOUND) {
+          LOGGER.debug("Restore status: {}", requestStatus);
+          printErrorMessage("Restore failed. ");
+          printResponseErrorMessages(requestStatusResponse);
+          return false;
+        }
+        try {
+          TimeUnit.SECONDS.sleep(1);
+        } catch (InterruptedException e) {
+          Thread.currentThread().interrupt();
```

Comment:
❓ @paouelle Is this a safe way to handle thread interrupts?

Reply:
https://stackoverflow.com/a/4906814

Reply:
The thing I am worried about is interrupts in loops. [This comment](https://github.com/codice/ddf/pull/2782) that @paouelle made explains a more simplified version. I'll try to explain it in my own words 😄:

When an interrupt occurs, it sets a flag on the thread that signifies there was an interrupt. The `try` block will check that flag occasionally, and if the flag is set we enter the `catch` for the with an `InterruptedException`.

The problem is that if we are looping over a `try`, and the `catch` of that try sets the interrupt flag using `Thread.currentThread().interrupt()`, then the `try` will be stuck in a loop of repeatedly 
1. checking the interrupted flag
2. catching `InterruptedException`
3. setting the interrupted flag within the `catch`

If that explanation didn't help, you should get some advice from Patrick 😃.

Reply:
Good catch Jacob, you are absolutely right. I commented below